x/inter-tx/types/msgs.go

```diff
@@ -52,3 +53,58 @@ func (msg MsgRegisterAccount) GetSigners() []sdk.AccAddress {
 	}
 	return []sdk.AccAddress{accAddr}
 }
+
+var _ sdk.Msg = &MsgSend{}
+
+// NewMsgSend creates a new MsgSend instance
+func NewMsgSend(
+	chainType, port, channel string, sender, toAddress sdk.AccAddress, amount []sdk.Coin,
+) *MsgSend {
+	return &MsgSend{
+		ChainType:     chainType,
+		SourcePort:    port,
+		SourceChannel: channel,
+		Sender:        sender,
+		ToAddress:     toAddress,
+		Amount:        amount,
+	}
+}
+
+// Route implements sdk.Msg
+func (MsgSend) Route() string {
+	return RouterKey
+}
+
+// Type implements sdk.Msg
+func (MsgSend) Type() string {
+	return TypeMsgSend
+}
+
+// GetSigners implements sdk.Msg
+func (msg MsgSend) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Sender}
+}
+
+// ValidateBasic performs a basic check of the MsgRegisterAccount fields.
+func (msg MsgSend) ValidateBasic() error {
+	if msg.Sender.String() == "" {
```

Comment:
```suggestion
	if strings.TrimSpace(msg.Sender.String()) == "" {
```

Reply:
ditto for ToAddress. Not sure if this is entirely necessary, I forget what the SDK set as the standard and why

Reply:
Updated.